### PR TITLE
Add dequant-scale squash

### PR DESCRIPTION
**Pull request by wozna**
This PR adds a squash dequantize-scale pattern. That squash removes scale operator and rewrites that scale to dequantize operator scale. Only works when the bias of the scale op is 0.
There are 12 such pairs in the Ernie model.

### CODE REVIEW

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass_tester.cc [hunk 1]
```diff
@@ -252,6 +257,20 @@ ProgramDesc BuildMultipleQuantizeProgramDesc(bool use_mkldnn, float first_scale,
   return prog;
 }
 
+// a->Dequant->b
+// b->Scale->c
+ProgramDesc BuildDequantScaleProgramDesc(bool use_mkldnn, float first_scale,
+                                         float second_scale, float bias) {
```
Comment (wojtuss) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass_tester.cc]: Please, rename the scale names into sth more meaningful, like `dequant_scale` and `scale_scale`.

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass.cc [hunk 2]
```diff
@@ -284,6 +284,49 @@ void CPUQuantizeSquashPass::MultipleQuantizeSquash(Graph* graph) const {
...
+        scale_op->Op()->GetAttrIfExists<float>("bias") == 0.0) {
+      auto dequant_scale = dequant_op->Op()->GetAttrIfExists<float>("Scale");
+      auto scale_scale = scale_op->Op()->GetAttrIfExists<float>("scale");
+
+      PADDLE_ENFORCE_NE(dequant_scale, 0.0f,
+                        platform::errors::InvalidArgument(
+                            "Dequantize scale should not be equal 0"));
+      PADDLE_ENFORCE_NE(scale_scale, 0.0f,
+                        platform::errors::InvalidArgument(
+                            "Scale of scale op should not be equal 0"));
```
Comment (wojtuss) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass.cc]: Please make sure the scales are strictly positive, not only non-zero.

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass_tester.cc [hunk 3]
```diff
@@ -544,6 +563,32 @@ TEST(CpuQuantizeSquashPass, quatize_with_different_scale) {
...
+  auto first_scale = 1.2345f;
+  auto second_scale = 1.5432f;
+  auto bias = 0.0f;
+  auto use_mkldnn = true;
+  // remove: dequant out, scale op
+  auto remove_nodes = 2;
+  CountNodeTest(
+      BuildDequantScaleProgramDesc(use_mkldnn, first_scale, second_scale, bias),
+      remove_nodes);
+}
```
Comment (wojtuss) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass_tester.cc:581]: Please, make sure the `dequantize`'s scale is correct after the squash.

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass_tester.cc [hunk 4]
```diff
@@ -544,6 +563,32 @@ TEST(CpuQuantizeSquashPass, quatize_with_different_scale) {
...
+  auto first_scale = 1.2345f;
+  auto second_scale = 1.5432f;
+  auto bias = 1.0f;
+  auto use_mkldnn = true;
+  // nothing change
+  auto remove_nodes = 0;
+  CountNodeTest(
+      BuildDequantScaleProgramDesc(use_mkldnn, first_scale, second_scale, bias),
+      remove_nodes);
+}
```
Comment (wojtuss) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass_tester.cc:597]: ditto

**Comment by luotao1** [comment]
Please cherry-pick into release/1.7

**Pull request merged**